### ISSUE TITLE
PR: "Export Type to IFC As" and "Type IFC Predefined Type" do not function 

### ISSUE THREAD

**Issue by canpolatbaris** [Autodesk/revit-ifc#635]
### Problem Description

What is happening?
Same results with IFC 4 and IFC 2x3 export
Also, all other elements in model does not export by defined entity and predefined type

![2023-05-13_13-51-01](https://github.com/Autodesk/revit-ifc/assets/110653439/a4857643-41f7-4798-af2c-a126fe64d9f1)
![2023-05-13_13-59-18](https://github.com/Autodesk/revit-ifc/assets/110653439/a672078f-5654-4968-b825-71a6931b4919)
![2023-05-13_13-57-40](https://github.com/Autodesk/revit-ifc/assets/110653439/070c2975-9cab-4da5-8b5a-dcff03cfd8c4)

### Revit Version

2024.0.x

### IFC for Revit Addon Version

24.x.x

### Windows Version

11 22H2

**Comment by AngelVelezSosa** [comment]
Is that for everything, just for that table?  Do it fail if you have a Revit file with just that table, and if so can you share just that file?

**Comment by canpolatbaris** [comment]
This is for everything, unfortunately, not for only the table. I will send a link for the file. Also, with different IFC exports, it creates the same results.

**Comment by canpolatbaris** [comment]
Can you send your email address, please?

**Comment by AngelVelezSosa** [comment]
My email address is angel.velez@autodesk.com.  That said, if this is truly a problem for all exports for you, then sending a Revit file with one item that fails is more useful than a large file.

**Comment by canpolatbaris** [comment]
I sent the native and IFC models @AngelVelezSosa

**Comment by canpolatbaris** [comment]
Also, the same problem with other Revit versions (2022)
![2023-05-16_16-45-43](https://github.com/Autodesk/revit-ifc/assets/110653439/7e0afcbf-94ff-4ee8-b37a-ded60570e40d)
![2023-05-16_16-45-14](https://github.com/Autodesk/revit-ifc/assets/110653439/671b5413-9d30-4b43-8426-dbfbd37e3b1e)
![2023-05-16_16-44-45](https://github.com/Autodesk/revit-ifc/assets/110653439/b3289d4f-92b2-404f-a5d1-5ed684e2dcbe)
![2023-05-16_16-44-14](https://github.com/Autodesk/revit-ifc/assets/110653439/bf19d1ff-b3b7-41d2-9937-41a2dac9c989)

Only the wall can be transferred right, I can change its IFC predefined type
![2023-05-16_16-47-33](https://github.com/Autodesk/revit-ifc/assets/110653439/6e6d3791-dbe6-472b-9499-f92fca9dfcf8)

**Comment by AngelVelezSosa** [comment]
@parrela can you reproduce this?  This isn't a common issue or else we'd have a lot of complaints (and failing tests).

**Comment by canpolatbaris** [comment]
yes, I created the last images with a new file to test if the problem persists with other versions or not.

**Comment by canpolatbaris** [comment]
Only the 2019 version of Revit works for me, I think it is about the IFC Exporter version, which I updated for other versions but 2019. You can find my IFC Exporter versions below:
For Revit 2019: IFC Exporter 19.5.1.0
For Revit 2022: IFC Exporter 22.6.2.0
For Revit 2024: IFC Exporter 24.1.0.22

**Comment by canpolatbaris** [comment]
A clean install of Revit did not solve the problem, also I tried each version of IFC Exporter. I also believe some installation (not installation but some Revit system file problem, because before IFC export was ok) situation.

Today I tried with another computer and it functions as expected. the problem is with my computer and Revit installation

**Comment by canpolatbaris** [comment]
Uninstalled all versions of all Autodesk products, and installed only Revit 2024, still same problem.

**Comment by parrela** [comment]
Wow, so maybe this is a display issue. are the graphics settings or hardware different between this one and the other machine? This might be a case for Autodesk Support. They might have more experience in dealing with stuff like this.

**Comment by ArtemHnatko** [comment]
The issue could not be reproduced in Revit 2024 & 2025.
The “Export Type to IFC As” and “Type IFC Predefined Type” parameters function as expected when exporting to IFC4 without enabling "Export user-defined property sets" or "Export parameter mapping table".
In the original report, these additional options were enabled, which may have contributed to the unexpected result.

**Testing environment:**
- Autodesk Revit 2024 (IFC 24.3.20.0)
- Autodesk Revit 2025 (IFC 25.4.0.27)

![Image](https://github.com/user-attachments/assets/1a5f620e-a983-4abe-8175-d94a2ac5e0f3)
